utils/static-device-names/files/static-device-names.init

```diff
@@ -0,0 +1,359 @@
+#!/bin/sh /etc/rc.common
+
+# This script is part of the static-device-names package.
+# See /usr/share/doc/static-device-names
+#
+# Copyright (C) 2024 qualIP Software.
+# Released under the GPL v3.0 or later.
+#
+# Location: /etc/init.d/static-device-names
+
+START=11
+NAME=static-device-names
+TRACE=${TRACE:-0}
+
+# don't run within buildroot
+[ -n "${IPKG_INSTROOT}" ] && return 0
+
+###################
+# LOCAL FUNCTIONS #
+###################
+
+N=$'\n'  # New line
+
+LOG() {
+  local level ; level=$1 ; shift
+  [ "$level" = "debug" ] && [ "$TRACE" = 0 ] && return
+  logger -s -t "$NAME" -p "syslog.$level" -- "$@"
+}
+
+target_devices=
+mac_to_device_mapping=
+pci_id_to_device_mapping=
+pci_slot_to_device_mapping=
+usb_slot_to_device_mapping=
+
+## Read configuration using UCI
+read_static_device_names_config() {
+  local rc=0
+  local dev
+  local mac
+  local pci_id
+  local pci_slot
+  local usb_slot
+  if [ -z "$target_devices" ] ; then
+    target_devices=
+    mac_to_device_mapping=
+    pci_id_to_device_mapping=
+    pci_slot_to_device_mapping=
+    usb_slot_to_device_mapping=
+    local i=0
+    while uci -q get "static-device-names.@device[$i]" > /dev/null ; do
+      dev=$(uci -q get "static-device-names.@device[$i].name")
+      if [ -n "$dev" ] ; then
+        target_devices="$target_devices$N$dev"
```

Comment:
What is this `$N`? I assume that this is a space character. Why do we use this? Can't we just use a space. It makes the source code unreadable.

Reply:
`N` was a clean way to add newline -- not a space. I renamed it to `NL` to be clearer.